Parser/Types/TypeAnalyze.cs

```diff
@@ -23,16 +25,29 @@ public override NType Visit (NProgram p)
    }
 
    public override NType Visit (NDeclarations d) {
-      Visit (d.Vars); return Visit (d.Funcs);
+      Visit (d.Consts); Visit (d.Vars); return Visit (d.Funcs);
+   }
+
+   public override NType Visit (NConstDecl c) {
+      if (!Has (c.Name)) mSymbols.Consts.Add (c);
+      else throw new ParseException (c.Name, $"'{c.Name.Text}' already defined.");
+      if (c.Expr != null) {
+         c.Expr.Accept (this);

```

Comment:
What if the expression uses some variables from an enclosing block? Their values will not be known at compile time (it's not a constant).